scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1822,8 +1827,26 @@ def matvec_XH_X(x):
 
     # Get a low rank approximation of the implicitly defined gramian matrix.
     # This is not a stable way to approach the problem.
-    eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
-                                  ncv=ncv, which=which, v0=v0)
+    if solver == 'lobpcg':
+
+        if which == 'LM':
+            largest = True
+        elif which == 'SM':
+            largest = False
+        else:
+            raise ValueError("which must be either 'LM' or 'SM'.")
+
+        if k == 1 and v0 is not None:
+            X = np.reshape(v0, (-1, 1))
+        else:
+            X = np.random.RandomState(52).randn(min(A.shape), k)
+
+        eigvals, eigvec = lobpcg(XH_X, X, tol=tol ** 2, maxiter=maxiter,
+                          largest=largest)
+
+    else:
+        eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
```

Comment:
see above for tolerance squaring

Reply:
I do not know if it is documented. I suppose that the reason might be that one measures the norm of the residual using the normal matrix A'*A. The serious issue is if the tolerance gets so small that the solver could never reach it.  I have not looked at this. It may be a good issue to investigate and possibly file, but I hope unrelated to this PR, since the unit tests pass...

Reply:
Please advise?

Reply:
I'd consider this issue separately.

Reply:
Thanks! No change, mark as resolved